client/components/Login/LoginComponent.js

```diff
@@ -4,17 +4,83 @@ import { Grid, Cell, Textfield, Button, Checkbox } from 'react-mdl';
 import Page from '../Page/PageComponent';
 
 export default class Login extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+      isUsernamePresent: false,
+      isPasswordPresent: false,
+      errorUsername: false,
+      errorPassword: false
```

Comment:
I'd personally advocate to keep the state minimal, so that we only have username and password. To display error in the correct case we can have `wasSubmitted` when submit is hit it's true. After a key enter it is done, set it false again.